lib/encryption/key_manager.dart

```diff
@@ -440,7 +441,9 @@ class KeyManager {
           'session_key': sess.outboundGroupSession!.session_key(),
         };
         try {
-          devicesToReceive.removeWhere((k) => !k.encryptToDevice);
+          devicesToReceive.removeWhere(
+            (k) => (!k.encryptToDevice && k.userId == client.userID!),
```

Comment:
Why don't we share the new keys with other users here? This is for sharing the own session with other users afterwards. I think what you actually want is, disable key requests->forwarding from other users. But not sharing keys for normal e2ee